src/lib/Server/Controller/Views.php

```diff
@@ -45,9 +46,9 @@ public function createView(Request $request)
         );
 
         if ($viewInput->query instanceof LocationQuery) {
-            $method = 'findLocations';
+            $method = [$this->searchService, 'findLocations'];
         } else {
-            $method = 'findContent';
+            $method = [$this->searchService, 'findContent'];
```

Comment:
_Side note: I'd really prefer here strategy-based approach. But I guess better this than nothing ;) We can be agile here and fix it step by step. +1_

Reply:
Yup. Adding a strategy would probably be the best approach, but then we would need to expand this change a lot: new tag, new services, new classes, new tests.

The goal is to improve without too much code and potential breaks.